src/main/java/edu/ucsb/mapache/controllers/SearchController.java

```diff
@@ -95,14 +95,17 @@
             you.setSearchRemain(100);
             you.setTime(currentTime);
         }
-    
+        if (you.getApiToken()!="invalid token")
+            apiToken = you.getApiToken();
+
         if(you.getSearchRemain()<=0){
             return searchQuotaExceeded();
         }
+        logger.info("The current api token is {}", apiToken);
```

Comment:
In practice, we shouldn't be logging API keys to console, so maybe consider removing this?

Reply:
sure. this was just for debugging purposes.